sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,325 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../../test/helpers";
+import { PartTimeMSFAAProcessResponseIntegrationScheduler } from "../msfaa-part-time-process-response-integration.scheduler";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFileFromStructuredRecords,
+  getStructuredRecords,
+  mockDownloadFiles,
+  MSFAA_PART_TIME_RECEIVE_FILE_WITH_CANCELATION_RECORD,
+  MSFAA_PART_TIME_RECEIVE_FILE_WITH_INVALID_RECORDS_COUNT,
+  MSFAA_PART_TIME_RECEIVE_FILE_WITH_INVALID_SIN_HASH_TOTAL,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import {
+  MSFAA_PART_TIME_MARRIED,
+  MSFAA_PART_TIME_OTHER_COUNTRY,
+  MSFAA_PART_TIME_RELATIONSHIP_OTHER,
+} from "./msfaa-part-time-process-integration.scheduler.models";
+import { saveMSFAATestInputsData } from "./msfaa-factory";
+import { In, IsNull } from "typeorm";
+
+describe(
+  describeProcessorRootTest(QueueNames.PartTimeMSFAAProcessResponseIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: PartTimeMSFAAProcessResponseIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let msfaaMocksDownloadFolder: string;
+
+    beforeAll(async () => {
+      msfaaMocksDownloadFolder = path.join(__dirname, "msfaa-receive-files");
+      // Set the ESDC response folder to the files mocks folders.
+      process.env.ESDC_RESPONSE_FOLDER = msfaaMocksDownloadFolder;
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(PartTimeMSFAAProcessResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any pending MSFAA.
```

Comment:
mm...here what does it mean "pending MSFAA"? when `dateSigned === null` then MSFAA is pending, right?